File: JAVA/EmberekOOP/src/main/java/com/example/Ember.java
package com.example;

import java.time.LocalDate;

public class Ember {
    private String nev;
    private String szulDatum;
    private String szulHely;
    private double tti; 

    public Ember(String nev, String szulDatum, String szulHely) {
        this.nev = nev;
        this.szulDatum = szulDatum;
        this.szulHely = szulHely;
        this.tti = Math.random() * 10 + 18;
    }

    public String getNev() {
        return nev;
    }

    public void setNev(String nev) {
        this.nev = nev;
    }

    public String getSzulDatum() {
        return szulDatum;
    }

    public void setSzulDatum(String szulDatum) {
        this.szulDatum = szulDatum;
    }

    public String getSzulHely() {
        return szulHely;
    }

    public void setSzulHely(String szulHely) {
        this.szulHely = szulHely;
    }

    public double getTti() {
        return tti;
    }

    public void setTti(double tti) {
        this.tti = tti;
    }

    public int getSzuletesiEv() {
        return Integer.parseInt(szulDatum.split("-")[0]);
    }

    public int getSzuletesiHonap() {
        return Integer.parseInt(szulDatum.split("-")[1]);
    }

    public int getSzuletesiNap() {
        return Integer.parseInt(szulDatum.split("-")[2]);
    }

    public int getEletkor() {
        int szuletesiEv = getSzuletesiEv();
        int aktualisEv = LocalDate.now().getYear();
        return aktualisEv - szuletesiEv;
    }

    @Override
    public String toString() {
        return "Név: " + nev + "\nSzületési dátum: " + szulDatum + "\nSzületési hely: " + szulHely + "\nTTI: " + tti;
    }
}

File: EmberekOOP/ember.cs
using System;

public class Ember
{
    private string nev;
    private string szulDatum;
    private string szulHely;
    private double testTomegIndex;

    // Paraméteres konstruktor
    public Ember(string nev, string szulDatum, string szulHely, double testTomegIndex)
    {
        this.nev = nev;
        this.szulDatum = szulDatum;
        this.szulHely = szulHely;
        this.testTomegIndex = testTomegIndex;
    }

    // Metódus a születési év visszaadására
    public int GetSzuletesiEv()
    {
        return int.Parse(szulDatum.Split('-')[0]);
    }

    // Metódus a születési hónap visszaadására
    public int GetSzuletesiHonap()
    {
        return int.Parse(szulDatum.Split('-')[1]);
    }

    // Metódus a születési nap visszaadására
    public int GetSzuletesiNap()
    {
        return int.Parse(szulDatum.Split('-')[2]);
    }

    // Metódus az életkor meghatározására
    public int GetEletkor()
    {
        int currentYear = DateTime.Now.Year;
        return currentYear - GetSzuletesiEv();
    }

    // Metódus a testtömeg index visszaadására
    public double GetTestTomegIndex()
    {
        return testTomegIndex;
    }

    // Tagolt ToString metódus
    public override string ToString()
    {
        return $"Név: {nev}\nSzületési dátum: {szulDatum}\nSzületési hely: {szulHely}\nTesttömeg index: {testTomegIndex:F2}";
    }
}